web-server/v0.4/src/pages/Summary/index.js

```diff
@@ -112,7 +111,48 @@ class Summary extends React.Component {
   };
 
   onTabChange = key => {
-    this.setState({ activeSummaryTab: key });
+    const { summaryTocResult } = this.props;
+    this.setState({
+      activeSummaryTab: key,
+      tocTree: summaryTocResult.tocResult,
+      originalTree: summaryTocResult.tocResult,
+    });
+  };
+
+  configChange = (value, category) => {
+    const { selectedConfig } = this.state;
+    if (value === undefined) {
+      delete selectedConfig[category];
+    } else {
+      selectedConfig[category] = value;
+    }
+    this.setState({ selectedConfig });
+  };
+
+  clearFilters = () => {
+    this.setState({
+      selectedConfig: [],
+    });
+  };
+
+  onload = tocTree => {
+    this.setState({
+      tocTree,
+    });
+  };
+
+  onSearchFile = value => {
```

Comment:
Searching for files doesn't seem to visualize the found file in the `TableTree` component. We should consider moving searching functionality to the `TableTree` component for better organization of helpers + state variables.